document.go

```diff
@@ -477,6 +516,29 @@ func extractDocumentData(rt reflect.Type, skipAssoc bool) documentData {
 	return data
 }
 
+// Check pointers to embedded structs for index path
+// Returns false if pointers are invalid and init is false
+func checkPointersForIndices(rv reflect.Value, indices []int, init bool) bool {
```

Comment:
deepsource is complaining because of the init parameter (countrol coupling). Should this really be split up? That'd end up with two almost identical functions

Reply:
I think we can ignore this